app/config/collections.php

```diff
@@ -1366,7 +1377,495 @@
             ],
         ],
     ],
- ];
+
+    'providers' => [
+        '$collection' => ID::custom(DATABASE::METADATA),
+        '$id' => ID::custom('providers'),
+        'name' => 'Providers',
+        'attributes' => [
+            [
+                '$id' => ID::custom('name'),
+                'type' => Database::VAR_STRING,
+                'format' => '',
+                'size' => 128,
+                'signed' => true,
+                'required' => true,
+                'default' => null,
+                'array' => false,
+                'filters' => [],
+            ],
+            [
+                '$id' => ID::custom('provider'),
+                'type' => Database::VAR_STRING,
+                'format' => '',
+                'size' => Database::LENGTH_KEY,
+                'signed' => true,
+                'required' => true,
+                'default' => null,
+                'array' => false,
+                'filters' => [],
+            ],
+            [
+                '$id' => ID::custom('type'),
+                'type' => Database::VAR_STRING,
+                'format' => '',
+                'size' => 128,
+                'signed' => true,
+                'required' => true,
+                'default' => null,
+                'array' => false,
+                'filters' => [],
+            ],
+            [
+                '$id' => ID::custom('default'),
+                'type' => Database::VAR_BOOLEAN,
+                'signed' => true,
+                'size' => 0,
+                'format' => '',
+                'filters' => [],
+                'required' => false,
+                'default' => false,
+                'array' => false,
+            ],
+            [
+                '$id' => ID::custom('credentials'),
+                'type' => Database::VAR_STRING,
+                'format' => '',
+                'size' => 16384,
+                'signed' => true,
+                'required' => true,
+                'default' => null,
+                'array' => false,
+                'filters' => ['json', 'encrypt'],
+            ]
+        ],
+        'indexes' => [
+            [
+                '$id' => ID::custom('_key_provider'),
+                'type' => Database::INDEX_KEY,
+                'attributes' => ['provider'],
+                'lengths' => [128],
+                'orders' => [Database::ORDER_ASC],
+            ],
+            [
+                '$id' => ID::custom('_key_name'),
+                'type' => Database::INDEX_FULLTEXT,
+                'attributes' => ['name'],
+                'lengths' => [128],
+                'orders' => [Database::ORDER_ASC],
+            ]
```

Comment:
Do we internally query on any other attributes in this collection? I'm thinking maybe `type`?